siac/main.go

```diff
@@ -93,6 +106,18 @@ func apiPost(call, vals string) (*http.Response, error) {
 		return nil, errors.New("no response from daemon")
 	}
 	// check error code
+	if resp.StatusCode == http.StatusUnauthorized {
+		resp.Body.Close()
+		// Prompt for password and retry request with authentication.
+		password, err := speakeasy.Ask("API password: ")
+		if err != nil {
+			return nil, err
+		}
+		resp, err = api.HttpPOSTAuthenticated("http://"+addr+call, vals, password)
+		if err != nil {
+			return nil, errors.New("no response from daemon - authentication failed")
+		}
+	}
```

Comment:
Yes this was my main reasoning for leaving non-sensitive API calls exposed. It's habbit for me to check up on Sia (make sure it's staying in sync, etc...) every once in a while using `siac` and it would be a pain to type the password every time.